eth/state_accessor.go

```diff
@@ -163,7 +163,7 @@ func (eth *Ethereum) hashState(ctx context.Context, block *types.Block, reexec u
 			return nil, nil, fmt.Errorf("processing block %d failed: %v", current.NumberU64(), err)
 		}
 		// Finalize the state so any modifications are written to the trie
-		root, err := statedb.Commit(current.NumberU64(), eth.blockchain.Config().IsEIP158(current.Number()), true)
+		root, err := statedb.Commit(current.NumberU64(), eth.blockchain.Config().IsEIP158(current.Number()))
```

Comment:
Where is the config for this to be set to true?

Reply:
Actually I have changed it to semantics using "false", and explicitly re-introduced the call to Reference.

This is more similar to upstream and is not needed to call reference atomically here because the triedb used for retracing is always one created in this function and passed back into it. (the chain triedb is only used in the readOnly case)

See this comment: https://github.com/ava-labs/subnet-evm/pull/1366#discussion_r1802271582